maven-plugin-api/src/main/java/org/apache/maven/plugin/logging/LogFactory.java

```diff
@@ -0,0 +1,62 @@
+package org.apache.maven.plugin.logging;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.plugin.logging.internal.ILogFactory;
+
+/**
+ * This interface supplies {@link Log} instances to Mojos. Plugin code may freely use this log factory to obtain
+ * loggers that will be abridged to Maven internal logging system.
+ *
+ * @since TBD
+ */
+public final class LogFactory
+{
+    private static ILogFactory bridge;
+
+    private LogFactory()
+    {
+        // no instances of this can be created
+    }
+
+    /**
+     * Initialized Mojo LogFactory with a bridge.
+     */
+    public static void initLogFactory( ILogFactory bridge )
```

Comment:
shouldn't be exposed IMO and we should ensure it is called very early in maven lifecycle + calling it twice with a different value should likely fail. Also means the log factory shouldnt call it in its constructor otherwise we can't change it after, no?

Reply:
reworked, this is not here anymore